File: baek10828/10828.js
//let fs = require('fs');
//let input = fs.readFileSync('/dev/stdin').toString().split('\n');

const input = [
  '14',
  'push', '1',
  'push', '2',
  'top',
  'size',
  'empty',
  'pop',
  'pop',
  'pop',
  'size',
  'empty',
  'pop',
  'push', '3',
  'empty',
  'top'
]
let num = [];
let s = 0;


for(i=1; i<input.length; i++){
  let now = input[i];
  let next = input[i+1];
  if(input[i]=='push'){
    num.push(next);
    s = s + 1;
  }
  if(now=='pop'){
    if(s>0){
      console.log(num.pop());
      s = s - 1;
    } else {
      console.log(-1)
    }
  }
  if(now=='size'){
    console.log(s);
  }
  if(now=='empty'){
    if(s>0){
      console.log(0)
    } else {
      console.log(1)
    }
  }
  if(now=='top'){
    if(s>0){
      console.log(num[s-1]);
    } else {
      console.log(-1);
    }
  }
}

File: baekjoon/10828.py
input = [
  '14',
  'push', '1',
  'push', '2',
  'top',
  'size',
  'empty',
  'pop',
  'pop',
  'pop',
  'size',
  'empty',
  'pop',
  'push', '3',
  'empty',
  'top'
]
num = []
s = 0
q = 0

for i in range(1,len(input)):
  now = input[i]
  if input[i]=='push':
    num.append(input[i+1])
    s = s + 1
  
  if now=='pop':
    if s>0:
      print(num[s-1])
      del num[s-1]
      s = s - 1;
    else:
      print(-1)
    
  if now=='size':
    print(s)
  
  if now=='empty':
    if s>0:
      print(0)
    else:
      print(1)
  
  if now=='top':
    if s>0:
      print(num[s-1])
    else:
      print(-1)